appinventor/components/src/com/google/appinventor/components/runtime/DataModel.java

```diff
@@ -0,0 +1,379 @@
+package com.google.appinventor.components.runtime;
+
+import com.github.mikephil.charting.data.Entry;
+import com.google.appinventor.components.runtime.util.ChartDataSourceUtil;
+import com.google.appinventor.components.runtime.util.YailList;
+import gnu.mapping.Symbol;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public abstract class DataModel<E> {
+  /**
+   * Local List of entries; The modifications of the Data are made
+   * directly to these Entries, which are meant to be detached from
+   * the Dataset object itself to prevent exceptions & crashes due
+   * to asynchronous operations.
+   */
+  protected List<E> entries;
+  /**
+   * Limit the maximum allowed real-time data entries
+   * Since real-time data comes in fast, the case of
+   * multi-data source input is unhandled since it's
+   * better to avoid it.
+   */
+  protected int maximumTimeEntries = 200;
+
+  public abstract void clearEntries();
```

Comment:
Move this abstract method down after the constructor, alphabetically sorted between the add and get methods.